utils/build/docker/python/parametric/apm_test_client/server.py

```diff
@@ -37,6 +38,10 @@
 
 spans: Dict[int, Span] = {}
 otel_spans: Dict[int, OtelSpan] = {}
+# Store the active span for each tracer in an array to allow for easy global access
+# FastAPI resets the contextvar containing the active span after each request
+active_ddspan = [None]
```

Comment:
Why a list and not just a direct assignment? Also could we just use `spans` and make sure we always append to it and then know the last span is always the active one? Or do we need to allow for something more nuanced?

Reply:
We can assign the `active_ddspan` to object but we will need to use the `global` keyword in every endpoint/function that modifies this field. If the global keyword is not used then the function will modify it's own copy of the global variable and this will introduce failures that are hard to debug. By using a list here we make setting/getting the active_span dummy/munir proof